server/game/core/cost/CostAdjuster.ts

```diff
@@ -41,14 +61,19 @@ export class CostAdjuster {
     public constructor(
         private game: Game,
         private source: Card,
-        properties: ICostAdjusterProperties,
-
-        /** @deprecated not implemented yet */
-        private penaltyAspect?: Aspect
+        properties: ICostAdjusterProperties
     ) {
+        this.costAdjustType = properties.costAdjustType;
+        if (this.costAdjustType === CostAdjustType.Increase || this.costAdjustType === CostAdjustType.Decrease) {
+            Contract.assertNotNullLike(properties.amount, 'Amount is required for Increase and Decrease.');
+        }
         this.amount = properties.amount || 1;
         this.costFloor = properties.costFloor || 0;
-        this.direction = properties.direction;
+        // TODO: Implement this further for Rey/Kylo
+        if (this.costAdjustType === CostAdjustType.IgnoreSpecificAspects) {
+            Contract.assertNotNullLike(this.ignoredAspects, 'Ignored Aspect list is required for IgnoreSpecificAspects');
```

Comment:
Couple modifications to this:

1. Also need to check that if it is a list that it is non-empty
2. We should validate that ignoredAspects does not have a value if costAdjustType is not IgnoreSpecificAspects